### PR TITLE
fix(modal): Fixed modal positioning for SUI v2.3

### PR DESCRIPTION
**Pull request by avaneev95**
Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/edcarroll/ng2-semantic-ui/blob/master/CONTRIBUTING.md#) guide.
 - [x] linted, built and tested the changes locally.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

This PR fixes the modal positioning for SUI v2.3 and adds `isCentered` property.
Additionally I've removed center content wrapper from the `SuiDimmer` since v2.3 dimmers with content now only need a single wrapping content container.
Also I had to create special SuiDimmer implementation for modals since the original SuiDimmer adds unwanted wrapping `div` which breaks modal's horizontal centering on small screens.

As mentioned in #361 there is a problem with Typescript version, so I've changed it to ">=2.4.2 <2.7".

### CODE REVIEW

**Comment by bmdatl** [comment]
is this going to be merged any time soon? this is an important fix for my app.

File: demo/src/app/pages/modules/modal/modal.page.ts [hunk 1]
```diff
@@ -97,10 +97,17 @@ export class ModalPage {
                     description: "Sets whether the modal displays against a light background.",
                     defaultValue: "false"
                 },
+                {
+                    name: "iCentered",
```
Comment (arutkowski00) [inline demo/src/app/pages/modules/modal/modal.page.ts]: @avaneev95 You've made a typo here, it should be `isCentered`

**Review by edcarroll — APPROVED** [review]

**Comment by edcarroll** [comment]
Thank you for this PR also 🎉

**Comment by icruces** [comment]
Is this going to be released any time soon?

**Pull request merged**